From 4de48e21bdce3c84e14bed84f250706e6c53a0e4 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 10 Oct 2025 11:30:03 +0200
Subject: [PATCH] ARM: configs: multi_v7_addons: fix display part for stm32mp1

Update drivers to set all of them as module drivers except DRM & SIMPLEDRM.

Change-Id: Ia6e9363002b5d42a150a93616f08463dd2c80be6
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/498338
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../configs/fragment-02-multiv7_addons.config   | 17 +++++------------
 1 file changed, 5 insertions(+), 12 deletions(-)

diff --git a/arch/arm/configs/fragment-02-multiv7_addons.config b/arch/arm/configs/fragment-02-multiv7_addons.config
index 855b8c59b298..47f62fe187ce 100644
--- a/arch/arm/configs/fragment-02-multiv7_addons.config
+++ b/arch/arm/configs/fragment-02-multiv7_addons.config
@@ -3,7 +3,8 @@ CONFIG_AEABI=y
 CONFIG_ARCH_FORCE_MAX_ORDER=12
 CONFIG_ARM_SMC_MBOX=y
 CONFIG_ARM_SMC_WATCHDOG=y
-CONFIG_BACKLIGHT_GPIO=y
+CONFIG_BACKLIGHT_GPIO=m
+CONFIG_BACKLIGHT_PWM=m
 CONFIG_CAN_M_CAN=y
 CONFIG_CAN_M_CAN_PLATFORM=y
 CONFIG_CHR_DEV_SG=y
@@ -17,16 +18,9 @@ CONFIG_CORESIGHT_STM=m
 CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
 CONFIG_DEBUG_PREEMPT=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
-CONFIG_DRM_DISPLAY_CONNECTOR=y
-CONFIG_DRM_I2C_ADV7511=y
-CONFIG_DRM_PANEL_HIMAX_HX8394=y
-CONFIG_DRM_PANEL_ORISETECH_OTM8009A=y
-CONFIG_DRM_PANEL_RAYDIUM_RM68200=y
-CONFIG_DRM_PANEL_ROCKTECH_HX8394=y
-CONFIG_DRM_SII902X=y
+CONFIG_DRM_PANEL_HIMAX_HX8394=m
+CONFIG_DRM_PANEL_SIMPLE=m
 CONFIG_DRM_SIMPLEDRM=y
-CONFIG_DRM_STM=y
-CONFIG_DRM_STM_DSI=y
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_FUTEX=y
 CONFIG_FW_LOADER_USER_HELPER=y
@@ -74,8 +68,7 @@ CONFIG_STM32_DBG_BUS=m
 CONFIG_STM32_IPCC=y
 CONFIG_STM32_RPROC=y
 CONFIG_TEE_REMOTEPROC=y
-CONFIG_TOUCHSCREEN_EDT_FT5X06=y
-CONFIG_TOUCHSCREEN_GOODIX=y
+CONFIG_TOUCHSCREEN_GOODIX=m
 CONFIG_USB_HIDDEV=y
 CONFIG_USB_ONBOARD_HUB=y
 CONFIG_USB_SERIAL=m
-- 
2.53.0

